moto/wafv2/models.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import unicode_literals
+
+import uuid
+
+from moto.core import BaseBackend, BaseModel
+from moto.wafv2.utils import make_webacl_arn
+
+US_EAST_1_REGION = "us-east-1"
+GLOBAL_REGION = "global"
+
+
+class WAFV2Backend(BaseBackend):
+    """
+    https://docs.aws.amazon.com/waf/latest/APIReference/API_Types_AWS_WAFV2.html
+    """
+
+    def __init__(self, region_name=None):
+        super(WAFV2Backend, self).__init__()
+        self.region_name = region_name
+        self.wacls = [WebACL("first-mock-webacl"), WebACL("second-mock-webacl")]
+
+    def reset(self):
+        region_name = self.region_name
+        self.__dict__ = {}
+        self.__init__(region_name)
+
+
+class WebACL(BaseModel):
+    def __init__(self, name="Mock-WebACL-name"):
+        self.Id = str(uuid.uuid4())
+        """
+        The name must have 1-128 characters. Valid characters: A-Z, a-z, 0-9, - (hyphen), and _ (underscore).
+        """
+        self.Name = "{0}-{1}".format(name, uuid.uuid4())[0:128]
+        self.ARN = make_webacl_arn(US_EAST_1_REGION, self.Name)
+        self.Capacity = 3
+        self.Description = "Mock WebACL named {0}".format(self.Name)
+        self.VisibilityConfig = VisibilityConfig(self.Name)
+        """
+        TODO: create classes for the below fields:
+        """
+        self.DefaultAction = {"Allow": {}}
+
+    def to_dict(self):
+        return {
+            "Id": self.Id,
+            "Name": self.Name,
+            "ARN": self.ARN,
+            "Capacity": self.Capacity,
+            "Description": self.Description,
+            "VisibilityConfig": self.VisibilityConfig.__dict__,
+        }
+
+
+class VisibilityConfig(BaseModel):
+    """
+    https://docs.aws.amazon.com/waf/latest/APIReference/API_VisibilityConfig.html
+    """
+
+    def __init__(self, metric_name):
+        self.SampledRequestsEnabled = True
+        self.CloudWatchMetricsEnabled = False
+        self.MetricName = metric_name
+
+
+wafv2_backends = {"global": WAFV2Backend(GLOBAL_REGION)}
```

Comment:
WAF is interesting, in that it combines both regional and global resources. As far as I know, it's the first service in Moto that does it.
The global/cloudfront resources are still persisted in `us-east-``, though. So the backends can still be created as every other service, having a separate backend per region. We'll just have to check for every call whether the scope was set to CLOUDFRONT or REGIONAL.

https://docs.aws.amazon.com/cli/latest/reference/wafv2/list-web-acls.html
>
> To work with CloudFront, you must also specify the Region US East (N. Virginia)

Reply:
The `get_available_regions` command does not return anything for `wafv2` at the moment, so let's use `waf-regional` for now. The list should be the same anyway.
```
wafv2_backends = {}
for region in Session().get_available_regions("waf-regional"):
    wafv2_backends[region] = WAFV2Backend(region)
for region in Session().get_available_regions("waf-regional", partition_name="aws-us-gov"):
    wafv2_backends[region] = WAFV2Backend(region)
for region in Session().get_available_regions("waf-regional", partition_name="aws-cn"):
    wafv2_backends[region] = WAFV2Backend(region)
```

Reply:
Thank you. Updated.